Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -119,10 +112,12 @@ private void SetCommonProperties(string eventName, object eventPayload, Activity
         {
             // activity name looks like 'Microsoft.Azure.ServiceBus.<Name>'
             // as namespace is too verbose, we'll just take the last node from the activity name as telemetry name
-            string[] activityNameSegments = activity.OperationName.Split('.');
-            if (activityNameSegments.Length > 0)
+            string activityName = activity.OperationName;
+            int lastDotIndex = activityName.LastIndexOf('.');
+            if (lastDotIndex > 0)
```

Comment:
If lastDotIndex is 0 (activityName has a leading period) or negative (activityName doesn't have a period), does `telemetry.Name` need to be set?

Reply:
this is invalid operation name, i.e. it should never ever happen.  I'd say there is no preference for this case. I'll put >= 0 for consistency (as 0 index does not lead to any exception or issue)